documentation/general/dotnet-run-file.md

```diff
@@ -203,10 +226,12 @@ Because these directives are limited by the C# language to only appear before th
 dotnet CLI can look for them via a regex or Roslyn lexer without any knowledge of defined conditional symbols
 and can do that efficiently by stopping the search when it sees the first "C# token".
 
-We do not limit these directives to appear only in entry point files.
-Indeed, it might be beneficial to let a non-entry-point file like `Util.cs` be self-contained and have all the `#:package`s it needs specified in it,
-which also makes it possible to share it independently or symlink it to multiple script folders.
-This is also similar to `global using`s which users usually put into a single file but don't have to.
+For a given `dotnet run file.cs`, we include directives from the current entry point file (`file.cs`) and all other non-entry-point files.
+The order in which other files are processed is currently unspecified (can change across SDK versions) but deterministic (stable in a given SDK version).
+We do not limit these directives to appear only in entry point files because it allows:
+- a non-entry-point file like `Util.cs` to be self-contained and have all the `#:package`s it needs specified in it,
+- which also makes it possible to share it independently or symlink it to multiple script folders,
+- and it's similar to `global using`s which users usually put into a single file but don't have to.
 
 We could consider deduplicating `#:` directives
```

Comment:
I'd at least expect us to block duplicate stuff, otherwise we've boxed ourselves into a corner we can't later design properly.

Reply:
We currently don't do any deduplication so all directives end up in the final project and MSBuild should fail for invalid duplicates.

Reply:
What duplicates do you expect MSBuild to fail on? ProjectReference maybe, but properties being assigned more than once is totally allowed.

Reply:
Right, I guess it should be simple to block duplicate directives for now, thanks.